### ISSUE TITLE
Give a try an AlphaGo-style value network

### ISSUE THREAD

**Issue by d180cf** [d180cf/dcnn-eval#9]
https://gogameguru.com/i/2016/03/deepmind-mastering-go.pdf

Page 27:

>Neural Network Architecture The input to the policy network is a 19 × 19 × 48 image stack
consisting of 48 feature planes. The first hidden layer zero pads the input into a 23 × 23 image,
then convolves k filters of kernel size 5×5 with stride 1 with the input image and applies a rectifier
nonlinearity. Each of the subsequent hidden layers 2 to 12 zero pads the respective previous hidden
layer into a 21×21 image, then convolves k filters of kernel size 3×3 with stride 1, again followed
by a rectifier nonlinearity. The final layer convolves 1 filter of kernel size 1 × 1 with stride 1, with
a different bias for each position, and applies a softmax function. The match version of AlphaGo
used k = 192 filters; Figure 2,b and Extended Data Table 3 additionally show the results of training
with k = 128, 256, 384 filters.

>The input to the value network is also a 19 × 19 × 48 image stack, with an additional binary
feature plane describing the current colour to play. Hidden layers 2 to 11 are identical to the policy
network, hidden layer 12 is an additional convolution layer, hidden layer 13 convolves 1 filter of
1 × 1 with stride 1, and hidden layer 14 is a fully connected linear layer with 256 rectifier units.
The output layer is a fully connected linear layer with a single tanh unit.
